### PR TITLE
feat: Anvil: Add `closeOnSubmit` property to modal widgets

### LINKED ISSUE #33240: [Bug]: Submit button issues on modal
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

- Submit button can be toggled off from the property pane but that does not clear `onSubmit` event from the list
- there's no difference between `onClose` and `onSubmit` behaviour - modal closes on submit so both events are triggered _together (this is when submit button is present on the widget)_

### Steps To Reproduce

1. DnD a modal widget and configure a custom `onSubmit` and `onClose` `showAlert` message
2. from preview mode, click on the submit button of modal and observe both `showAlert` message showing up
3. switch to edit mode and notice turning off the submit toggle does not remove the `onSubmit` option from events

### Public Sample App

_No response_

### Environment

Deploy Preview

### Severity

High (Blocker to building or releasing)

### Issue video log

_No response_

### Version

cloud

### PR DESCRIPTION
**Pull request by riodeuno**
## Description
- Creates a new property control for Modal widgets. `closeOnSubmit`. 
  - This property (default `true`) configures if the modal must close on the submit button click.

Fixes #33240 

## Automation

/ok-to-test tags="@tag.Anvil"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9299980305>
> Commit: 59fc2957f0b4ff3c47788480b0d82adf99596265
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9299980305&attempt=1" target="_blank">Click here!</a>

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

### CODE REVIEW

**Review by marks0351 — DISMISSED** [review]

**Comment by riodeuno** [comment]
Thanks @marks0351.

@KelvinOm I'd like for you to take a look at the changes in the Modal components and stories. I'm not sure how we write tests for these components today. I've forced the value of `closeOnSubmit` to be `true` in all cases, so nothing should break. But, I'd like to know if I need to create any additional tests or stories to showcase this feature.

File: app/client/packages/design-system/widgets/src/components/Modal/stories/ModalExamples.tsx [hunk 1]
```diff
@@ -27,6 +27,7 @@ export const ModalExamples = () => {
   const smallRef = useRef(null);
   const mediumRef = useRef(null);
   const largeRef = useRef(null);
+  const closeOnSubmit = true;
```
Comment (KelvinOm) [inline app/client/packages/design-system/widgets/src/components/Modal/stories/ModalExamples.tsx:30]: I think you can delete all changes in this file now.

File: app/client/packages/design-system/widgets/src/components/Modal/src/ModalFooter.tsx [hunk 2]
```diff
@@ -20,7 +21,7 @@ export const ModalFooter = (props: ModalFooterProps) => {
       setIsLoading(true);
       await onSubmit();
       setIsLoading(false);
-      setOpen(false);
+      if (closeOnSubmit ?? false) setOpen(false);
```
Comment (KelvinOm) [inline app/client/packages/design-system/widgets/src/components/Modal/src/ModalFooter.tsx:24]: if (closeOnSubmit) setOpen(false);

File: app/client/packages/design-system/widgets/src/components/Modal/src/ModalFooter.tsx [hunk 3]
```diff
@@ -7,6 +7,7 @@ import type { ModalFooterProps } from "./types";
 
 export const ModalFooter = (props: ModalFooterProps) => {
   const {
+    closeOnSubmit,
```
Comment (KelvinOm) [inline app/client/packages/design-system/widgets/src/components/Modal/src/ModalFooter.tsx:10]:
```suggestion
    closeOnSubmit = true,
```

**Review by KelvinOm — APPROVED** [review]

**Pull request merged**